src/main/resources/xslt/preservica2solr.xsl

```diff
@@ -0,0 +1,178 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<xsl:transform xmlns:t="http://www.test.com/"
+               xmlns:m="http://www.loc.gov/mods/v3"
+               xmlns:mets="http://www.loc.gov/METS/"
+               xmlns:tei="http://www.tei-c.org/ns/1.0"
+               xmlns:f="http://www.w3.org/2005/xpath-functions"
+               xmlns:h="http://www.w3.org/1999/xhtml"
+               xmlns:xs="http://www.w3.org/2001/XMLSchema"
+               xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+               xmlns:my="urn:my"
+               xmlns:premis="http://www.loc.gov/premis/v3"
+               xmlns:mix="http://www.loc.gov/mix/v20"
+               xmlns:pbc="http://www.pbcore.org/PBCore/PBCoreNamespace.html"
+               xmlns:xip="http://example.com/"
+               version="3.0">
+
+  <xsl:output method="text" />
+  <!-- Declare all externally provided parameters here -->
+  <!-- Provided by the Java code that calls the Transformer. (Key,Value) pairs are given in a map -->
+  <xsl:param name="access_blokeret"/>
+  <xsl:param name="access_pligtafleveret"/>
+  <xsl:param name="access_ejermaerke"/>
+  <xsl:param name="access_note"/>
+  <xsl:param name="access_skabelsesaar"/>
+  <xsl:param name="access_ophavsperson_doedsaar"/>
+  <xsl:param name="access_searlige_visningsvilkaar"/>
+  <xsl:param name="access_materiale_type"/>
+  <xsl:param name="access_foto_aftale"/>
+  <xsl:param name="access_billede_aftale"/>
+  <xsl:param name="access_ophavsret_tekst"/>
+  <xsl:param name="imageserver"/>
+  <xsl:param name="recordID"/>
+
+  <xsl:template match="/" >
+    <xsl:variable name="solrjson">
+      <f:map>
+
+        <f:string key="id">
+          <xsl:value-of select="$recordID"/>
+        </f:string>
+        <!-- Not sure if this accession ref is correctly understood-->
+        <f:string key="accession_number">
+          <xsl:value-of select="/xip:DeliverableUnit/AccessionRef"/>
+        </f:string>
+        <f:string key="collection">
+          <xsl:value-of select="/xip:DeliverableUnit/Metadata/pbc:PBCoreDescriptionDocument/pbcoreInstantiation/formatLocation"/>
```

Comment:
Instead of doing a lot of long matches like `/xip:DeliverableUnit/Metadata/pbc:PBCoreDescriptionDocument/pbcoreInstantiation/formatLocation` maybe create a template that matches on `/xip:DeliverableUnit/Metadata/pbc:PBCoreDescriptionDocument/` and process the PBCore elements in there with just `pbcoreInstantiation/formatLocation`? That would also open up for later refactoring to a generic PBCore-handling helper XSLT.